src/Autofac/Core/Activators/Reflection/NoPublicConstructorException.cs

```diff
@@ -0,0 +1,59 @@
+﻿// This software is part of the Autofac IoC container
+// Copyright � 2011 Autofac Contributors
+// http://autofac.org
+//
+// Permission is hereby granted, free of charge, to any person
+// obtaining a copy of this software and associated documentation
+// files (the "Software"), to deal in the Software without
+// restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+// OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using System.Globalization;
+
+namespace Autofac.Core.Activators.Reflection
+{
+    public class NoPublicConstructorException : Exception
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NoPublicConstructorException"/> class.
+        /// </summary>
+        /// <param name="type">The type with no public constructor but was registered.</param>
+        public NoPublicConstructorException(Type type)
+            : base(FormatMessage(type))
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NoPublicConstructorException"/> class.
+        /// </summary>
+        /// <param name="type">The type with no public constructor but was registered.</param>
+        /// <param name="innerException">The inner exception.</param>
+        public NoPublicConstructorException(Type type, Exception innerException)
```

Comment:
We probably need a couple of additional constructors:

- Type and message
- Type, message, and inner exception

That way a custom message could be put in to override the default.